modules/main_collector.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import traceback
+
+from prometheus_client import Metric, REGISTRY, start_http_server
+from modules.ucsm import UcsmServer
+from modules.server_license import UcsServerLicense
+
+class MainCollector(object):
+    def __init__(self, servers, creds):
+        self.servers = servers
+        self.creds = creds
+        self.handles = dict()
+
+    def __del__(self):
+        print("logging out of handles")
+        # for x in range(self.handles):
+        #     if len(self.handles)>x:
+        #         self.handles[x].logout()
+
```

Comment:
Add a describe() function to avoid prometheus running collect() initially when the the Collector is registered. See our other exporters.